contracts/eoslib/string.h

```diff
@@ -0,0 +1,10 @@
+/**
+ *  @file
+ *  @copyright defined in eos/LICENSE.txt
+ */
+#pragma once
+#include <eoslib/types.h>
+
+extern "C" {
+    uint32_t is_valid_utf8_str( const char* str, uint32_t len);
```

Comment:
I prefer for this to be "assert_is_utf8" and have it throw an exception on failure. I cannot think of a reason why someone would need "if not utf8 then do..." type logic and any "assert..." type APIs can be optimized out during replay.

Reply:
Right, I think this makes more sense.